nimbus : fix dark-nimbus and light-nimbus theme

jwntree opened the pull request:
Fixes nimbus-light and nimbus-dark themes not working properly due to missing gtk 3.0 folder.

There are some unexpected changes.  I can't sure if this is the right change.

- Added some build dependencies.

- path=usr/lib/gtk-2.0/2.10.0/engines/libnimbus.so is deleted from nimbus.p5m. It seems like the 32bit dependency is gone.
- Replaced gnome-autogen.sh to mate-autogen. There is a stupid automake version check in gnome-autogen.sh and automake version was 1.16. I considered patching gnome-common to support 1.16, but It seems more common to use mate-common instead.
- Modified some makefile.am for make install to ensure the path is set properly.

AndWac commented:
You rightfully have dropped 32 bit. This package would need some libraries in 32 bit we don't have anymore.

I have the impression that you didn't run gmake sample-manifest first. This would have changed manifest/sample-manifest.p5m. We use theses sample-manifest files in order to have a template for the real manifest file(s) and to get an idea of the needed changes when anything changes.

You should also run gmake REQUIRED_PACKAGES which will automatically detect many dependencies. This adds a special comment line in front of them. You should not edit these manually. Alas not all dependencies can be detected so sometimes additional REQUIRED_PACKAGE lines need to be added manually. We separate the manually added lines in order to keep them intact when gmake REQUIRED_PACKAGES will be run.

After gmake publish the pkg5 file should also change if any change has happened to the REQUIRED_PACKAGES entries.

Please update sample-manifest.p5m and pkg5 and add them to your PR.

iigs commented:
It is not recommended to add patches here to change the nimbus theme. Instead, create a PR with the appropriate changes at https://github.com/OpenIndiana/nimbus . Once the PR is accepted in OpenIndiana/nimbus repo, then patches are not needed here.  Do you want to consider integrating more from https://github.com/RocketMan/slim-nimbus into OpenIndiana/nimbus ?  My recommendation is not to include any of GTK 4.0 since it is not complete enough for many gtk4 apps.

jwntree commented:
I submitted a PR to the original repository instead patches and removed the dependencies I had manually added.
The build may fail due to the automake version check in gnome-autogen.sh.
(I removed Replaced gnome-autogen.sh to mate-autogen patch too.)

AndWac commented on components/desktop/nimbus/Makefile:
```diff
@@ -53,13 +53,6 @@ CONFIGURE_SCRIPT=$(SOURCE_DIR)/autogen.sh
-
-test:		$(NO_TESTS)
-
 REQUIRED_PACKAGES += library/desktop/gdk-pixbuf
```
For automake you can add the following:
\# Manually added build dependencies
REQUIRED_PACKAGES += developer/build/automake-115

Alas there seems to be at least one more problem when I run gmake sample-manifest here:
(cd /export/home/andreas/oi-userland/components/desktop/nimbus/build/prototype/i386/usr/share/icons/nimbus/12x12 && /usr/lib/icon-name-mapping -c actions -e /export/home/andreas/oi-userland/components/desktop/nimbus/build/amd64/icons/my-mapping.xml)
Can't locate XML/Simple.pm in @INC (you may need to install the XML::Simple module) (@INC entries checked: /usr/perl5/site_perl/5.40/i86pc-solaris-thread-multi-64 /usr/perl5/site_perl/5.40 /usr/perl5/vendor_perl/5.40/i86pc-solaris-thread-multi-64 /usr/perl5/vendor_perl/5.40 /usr/perl5/5.40/lib/i86pc-solaris-thread-multi-64 /usr/perl5/5.40/lib) at /usr/lib/icon-name-mapping line 12.
BEGIN failed--compilation aborted at /usr/lib/icon-name-mapping line 12.
make[4]: *** [Makefile:535: install-data-local] Error 2
make[4]: Leaving directory '/export/home/andreas/oi-userland/components/desktop/nimbus/build/amd64/icons/12x12/actions'
make[3]: *** [Makefile:426: install-am] Error 2
make[3]: Leaving directory '/export/home/andreas/oi-userland/components/desktop/nimbus/build/amd64/icons/12x12/actions'
make[2]: *** [Makefile:387: install-recursive] Error 1
make[2]: Leaving directory '/export/home/andreas/oi-userland/components/desktop/nimbus/build/amd64/icons/12x12'
make[1]: *** [Makefile:447: install-recursive] Error 1
make[1]: Leaving directory '/export/home/andreas/oi-userland/components/desktop/nimbus/build/amd64/icons'
make: *** [Makefile:503: install-recursive] Error 1
gmake: *** [/export/home/andreas/oi-userland/make-rules/common-rules.mk:67: /export/home/andreas/oi-userland/components/desktop/nimbus/build/amd64/.installed] Error 2

AndWac commented:
When you are able to run gmake publish in your environment you need to run
gmake REQUIRED_PACKAGES
and check the Makefile again. This should insert a line
\# Auto-generated dependencies
followed by a list of REQUIRED_PACKAGE statements.
Remove redunant lines from above (manually added dependencies).

jwntree committed:
nimbus : fix dark-nimbus and light-nimbus theme

AndWac commented:
Thanks for your contribution. I hope we will see more from you in the future. Any helping hand is welcome!
